pkg/cli/cmd/deploy/deploy.go

```diff
@@ -183,17 +187,17 @@ func (r *Runner) Validate(cmd *cobra.Command, args []string) error {
 	if err != nil {
 		return err
 	}
-	env, err := client.GetEnvironment(cmd.Context(), r.EnvironmentName)
+	env, err := client.GetEnvironment(cmd.Context(), r.EnvironmentNameOrID)
 	if err != nil {
 		// If the error is not a 404, return it
 		if !clients.Is404Error(err) {
 			return err
 		}
 
-		// If the environment doesn't exist, but the user specified it as
+		// If the environment doesn't exist, but the user specified its name or resource id as
 		// a command-line option, return an error
 		if cli.DidSpecifyEnvironmentName(cmd, args) {
-			return clierrors.Message("The environment %q does not exist in scope %q. Run `rad env create` first.", r.EnvironmentName, r.Workspace.Scope)
+			return clierrors.Message("The environment %q does not exist in scope %q. Run `rad env create` first. You could also provide the environment ID if the environment exists in a different group.", r.EnvironmentNameOrID, r.Workspace.Scope)
```

Comment:
Can we only provide an **environmentID** when we are deploying to another group?

Reply:
Yes. If not, it will enter the codeblock `if cli.DidSpecifyEnvironmentName` and ppint the error message indicating the environment isnt found in group and we should specify the id if it exists in a different group.